pkg/netpol/eval/ingressanalyzer/ingress_analyzer.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2022
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ingressanalyzer
+
+import (
+	ocroutev1 "github.com/openshift/api/route/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/np-guard/netpol-analyzer/pkg/netpol/eval"
+	"github.com/np-guard/netpol-analyzer/pkg/netpol/eval/internal/k8s"
+	"github.com/np-guard/netpol-analyzer/pkg/netpol/scan"
+)
+
+type (
+	IngressAnalyzer struct {
+		routesMap map[string]map[string]*k8s.Route // map from namespace to map from route name to its object
+		// k8sIngressMap map[string]map[string]*k8s.Ingress // map from namespace to map from ingress name to its object
+	}
+
+	// NotificationTarget defines an interface for updating the state needed for ingress
+	// decisions
+	NotificationTarget interface {
```

Comment:
In order to avoid code duplication (with `PolicyEngine`), we may move this to an eval "utility" package to be used in both structs. (along with general implementations of the `Upsert ` and `Delete` funcs)